src/client/opamCliMain.ml

```diff
@@ -132,8 +132,12 @@ let check_and_run_external_commands () =
     else
     (* No such command, check if there is a matching plugin *)
     let command = OpamPath.plugin_prefix ^ name in
-    let answer = if yes then Some true else OpamCoreConfig.E.yes () in
-    OpamCoreConfig.init ~answer ();
+    let answer =
+      if yes then Some `all_yes else
+        OpamStd.Option.Op.(OpamCoreConfig.E.yes ()
+                           >>| (fun b -> if b then `all_yes else `all_no))
+    in
+    OpamCoreConfig.init ?answer ();
```

Comment:
Shouldn't this take `OPAMCONFIRMLEVEL` and `OPAMNO` into account as well?

Reply:
Even if you can't say `opam --confirm-level=yes plugin blah`, I think it would be correct to expect `OPAMCONFIRMLEVEL=yes opam plugin blah` to install the plugin automatically and `OPAMNO=true opam plugin blah` not to?

Reply:
It should yes. For that we need to have `OPAMCONFIRMLEVEL` and `OPAMNO` out of the environment variables cli versioning system, and initialised at [launch](https://github.com/ocaml/opam/blob/master/src/client/opamArg.ml#L42).

Reply:
In fact, it is possible here to initialise environment variables, as we have cli, and it's a piece of code that will just install or launch the plugin. I think that it is reasonable that plugins inherit of environment variable set?

Reply:
Agreed!